pyat/at/lattice/lattice_object.py

```diff
@@ -263,29 +262,19 @@ def __str__(self):
 
     def __add__(self, elems):
         """Add elems, an iterable of AT elements, to the lattice"""
-        newring = Lattice(self)
-        newring.extend(elems)
-        return newring
+        return self.concatenate(elems, copy=True)
 
     def __iadd__(self, elems):
-        return super(Lattice, self).__iadd__(self._addition_filter(elems))
+        self.concatenate(elems, copy=False)
+        return self
 
     def __mul__(self, n):
-        """Repeats n times the lattice"""
-        periodicity = self.periodicity
-        if n != 0 and periodicity > 1:
-            nbp = periodicity / n
-            periodicity = int(round(nbp))
-            if abs(periodicity - nbp) > _TWO_PI_ERROR:
-                warn(AtWarning('Non-integer number of cells: {}/{}. Periodi'
-                               'city set to 1'.format(self.periodicity, n)))
-                periodicity = 1
-        # noinspection PyTypeChecker
-        return Lattice(elem_generator,
-                       itertools.chain(*itertools.repeat(self, n)),
-                       iterator=self.attrs_filter, periodicity=periodicity)
+        return self.repeat(n)
 
-    def _addition_filter(self, elems: Iterable[Element]):
+    def __reversed__(self):
+        return self.reverse(copy=True)
```

Comment:
This returns  a `Lattice` object, while it should return a "reverse iterator": iterator over the elements in reversed order:
```python
for elem in self[::-1]:
    yield elem.copy.swapfaces()
```

Or should the copy be done in the `swapfaces()` method?